libsolidity/formal/SMTChecker.cpp

```diff
@@ -605,13 +608,13 @@ void SMTChecker::checkBooleanNotConstant(Expression const& _condition, string co
 		return;
 
 	m_interface->push();
-	m_interface->addAssertion(expr(_condition));
-	auto positiveResult = checkSatisifable();
+	m_interface->addAssertion(currentPathConditions() && expr(_condition));
```

Comment:
What happens in case `currentPathCondition()` itself is not satisfiable, i.e. in case we use `if (x)` in unreachable code? We should display a different error message to the user in this case.

Reply:
In that case, some previous condition would already have triggered the unreachable code message, wouldn't it?

Reply:
Sure, but it means that we get an error message that is not really meaningful. Users will probably not understand the fact that the condition is constant just because it is not reachable.

Reply:
I don't really get which message would be the problem currently.
I just ran the example `if (a == 0) if (a > 0) if (a > 2) {}` and the output is:
```
unre.sol:6:21: Warning: Condition is always false.
    if (a == 0) if (a > 0) if (a > 2) {}
                    ^---^
unre.sol:6:32: Warning: Condition unreachable.
    if (a == 0) if (a > 0) if (a > 2) {}
                               ^---^
```
Do you think telling the user that it's unreachable doesn't really work?

Reply:
Ah I'm sorry. I didn't see unreachability is actually checked in the same function. It's fine! :-)

Reply:
Ok cool :)